### PR TITLE
Checking go version in scripts

### PR DESCRIPTION
**Pull request by alxsap**

### CODE REVIEW

**Comment by alxsap** [comment]
Don't get it. Same go version, GOPATH set, etc. locally. There it works.
```
Installing mockgen
go: cannot use path@version syntax in GOPATH mode
Installation of mockgen failed
script returned exit code 1
```

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -115,7 +132,7 @@ GOPATH_1=${GOPATH%%:*}  # the first entry of the GOPATH
 MOCKGEN_EXE="$GOPATH_1/bin/mockgen"
 if [[ ! -x $MOCKGEN_EXE ]]; then
     echo "Installing mockgen"
-    ( cd "$GOPATH_1" && go get github.com/golang/mock/mockgen ) || die "Installation of mockgen failed"
+    ( cd "$GOPATH_1" && go get github.com/golang/mock/mockgen@v1.3.1 ) || die "Installation of mockgen failed"
```
Comment (alxsap) [inline hack/update-codegen.sh:135]: @version does not work

Comment (alxsap) [inline hack/update-codegen.sh:135]:
```suggestion
    ( cd "$GOPATH_1" && go get github.com/golang/mock/mockgen ) || die "Installation of mockgen failed"
```

File: build.sh [hunk 2]
```diff
@@ -12,6 +12,16 @@ function banner1() {
     echo $'\n'"===" "$@" $'\n'
 }
 
+function checkGoVersion() {
+    local expectedVersion=`cat GOLANG_VERSION`
```
Comment (anbrsap) [inline build.sh:16]: Combining `local` with assignment via command substitution has the problem that an error in the command substitution is not handled. Split variable declaration and assignments:

    local myvar
    myvar=$(echo blah)

File: build.sh [hunk 3]
```diff
@@ -12,6 +12,16 @@ function banner1() {
...
+    [[ ! -z $expectedVersion ]] || die
+    go version | grep "${expectedVersion}"
```
Comment (anbrsap) [inline build.sh:18]: Not sufficient: expectedVerion=1.13.1 would match 1.13.10 for example.

File: hack/update-codegen.sh [hunk 4]
```diff
@@ -83,12 +83,27 @@ function generate_mocks() {
...
+
+function checkGoVersion() {
+    local expectedVersion=`cat $HERE/../GOLANG_VERSION`
+    [[ ! -z $expectedVersion ]] || die
+    go version | grep "${expectedVersion}"
+    local result=$?
+    if [[ $result != 0 ]]; then
+        die "error: Expected Go version ${expectedVersion} but was: $(go version)"
+    fi
+}
```
Comment (anbrsap) [inline hack/update-codegen.sh:105]: dito

File: build.sh [hunk 5]
```diff
@@ -12,6 +12,16 @@ function banner1() {
     echo $'\n'"===" "$@" $'\n'
 }
 
+function checkGoVersion() {
+    local expectedVersion=`cat GOLANG_VERSION`
+    [[ ! -z $expectedVersion ]] || die
```
Comment (anbrsap) [inline build.sh:17]: Simpler:

```go
[[ $expectedVersion ]] || die
```

**Pull request merged**